security/tpm/test/_tpm_test.js

```diff
@@ -2,45 +2,151 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 'use strict';
-
+var common = require("azure-iot-common");
 var TpmSecurityClient  = require('../lib/tpm').TpmSecurityClient ;
+var tss = require("tss.js");
+var sinon = require('sinon');
+
 var assert = require('chai').assert;
 
-describe('tpm', function () {
-  this.timeout(1000);
+var fakeSimpleTpmClient = {
+  allowErrors: () => {return this},
+  connect: (callback) => {callback();},
+  ReadPublic: (handle, callback) => {
+    callback(null,null);
+  },
+  fakeSimpleTpmClient: () => {},
+  getLastResponseCode: () => {
+    return tss.TPM_RC.SUCCESS;
+  }
+
+}
+function testFalsyArg(methodUnderTest, argName, argValue, ExpectedErrorType) {
+  var errorName = ExpectedErrorType ? ExpectedErrorType.name : 'Error';
+  it('Throws a ' + errorName + ' if \'' + argName + '\' is \'' + JSON.stringify(argValue) + '\' (type:' + typeof(argValue) + ')', function() {
+    var client = new TpmSecurityClient(null, fakeSimpleTpmClient);
+    assert.throws(function() {
+      client[methodUnderTest](argValue, function() {});
+    }, ExpectedErrorType);
+  });
+}
+
 
-  var obj = new TpmSecurityClient ();
+describe('tpm', function () {
+  this.timeout(5000);
```

Comment:
>this.timeout(5000); [](start = 2, length = 19)

why?

Reply:
Total time for all of the unit tests to run?  I suppose I could crank it back down to 1000.

---
In reply to: [160304244](https://github.com/Azure/azure-iot-sdk-node/pull/185#discussion_r160304244) [](ancestors = 160304244)

Reply:
that is not the total time, the timeout value applies to each test individually. the default is 2000. Leaving it to default is usually a good thing - unless the unit tests blocks on I/O and requires a longer timeout.

---
In reply to: [160484920](https://github.com/Azure/azure-iot-sdk-node/pull/185#discussion_r160484920) [](ancestors = 160484920,160304244)